lib/base.js

```diff
@@ -250,9 +250,46 @@ class ConnectionPool extends EventEmitter {
           return
         }
 
+        let createAttempt = 0
+
+        function create () {
+          const createId = createAttempt++
+          let timeout = this.config.pool ? this.config.pool.acquireTimeoutMillis : undefined
```

Comment:
We're going to need some docs on how to specify this.

I don't think the timeout should be infinite by default as none of this useful work would actually get executed. Maybe 32s (4 retry attempts) is sensible?

Reply:
I don't think we should be changing the default values. It's up to devs to define their acquireTimeouts and some people seem to want a never-ending acquire timeout...

Reply:
Let's revisit this once in the wild.